observationtree/src/main/scala/explore/observationtree/ConstraintSetObsList.scala

```diff
@@ -0,0 +1,383 @@
+      observationsView.mod(obsCsAdjuster.set(csIdOpt)) >>
+        // 2) Send mutation
+        csIdOpt.map(newId => moveObs(obsId, newId)).orEmpty
+
+    }
+
+    protected def onDragEnd(
+      setter:      Undoer.Setter[IO, ConstraintSetsWithObs],
+      expandedIds: View[SortedSet[ConstraintSet.Id]]
+    )(implicit
+      c:           TransactionalClient[IO, ObservationDB]
+    ): (DropResult, ResponderProvided) => IO[Unit] =
+      (result, _) =>
+        $.propsIn[IO] >>= { props =>
+          result.destination.toOption
+            .map(destination =>
+              (Observation.Id.parse(result.draggableId) match {
+                case Some(obsId) =>
+                  val obsWithId: GetAdjust[ObsList, obsListMod.ElemWithIndex] =
+                    obsListMod.withKey(obsId)
+
+                  val set: Option[Option[ConstraintSet.Id]] => IO[Unit] =
+                    setter.set[Option[Option[ConstraintSet.Id]]](
+                      props.constraintSetsWithObs.get,
+                      getConstraintSetForObsWithId(obsWithId.getter).get,
+                      setConstraintSetForObsWithId(props.constraintSetsWithObs, obsId, obsWithId)
+                    )
+
+                  destination.droppableId match {
+                    case UnassignedObsId           => set(none.some)
+                    case ConstraintSet.Id(newCsId) =>
+                      expandedIds.mod(_ + newCsId) >> set(newCsId.some.some)
+                    case _                         => IO.unit
+                  }
+                case None        => IO.unit
+              })
+            )
+            .orEmpty
+        }
+
+    def toggleExpanded(
+      id:          ConstraintSet.Id,
+      expandedIds: View[SortedSet[ConstraintSet.Id]]
+    ): Callback =
+      expandedIds
+        .mod(expanded => expanded.exists(_ === id).fold(expanded - id, expanded + id))
+        .runAsyncCB
+
+    // Adapted from https://github.com/atlassian/react-beautiful-dnd/issues/374#issuecomment-569817782
+    def getDraggedStyle(style:   TagMod, snapshot: Draggable.StateSnapshot): TagMod =
+      if (!snapshot.isDragging)
+        TagMod.empty
+      else if (!snapshot.isDropAnimating)
+        style
+      else
+        TagMod(style, ^.transitionDuration := "0.001s")
+
+    def getListStyle(isDragging: Boolean): TagMod =
+      ExploreStyles.DraggingOver.when(isDragging)
+
+    def render(props: Props, state: State): VdomElement = AppCtx.withCtx { implicit ctx =>
+      val observations       = props.constraintSetsWithObs.get.obs
+      val obsByConstraintSet = observations.toList.groupBy(_.constraintSetId)
+
+      val constraintSets        = props.constraintSetsWithObs.get.constraintSets
+      val constraintSetsWithIdx = constraintSets.toList.zipWithIndex
+
+      val unassignedObs = obsByConstraintSet.get(none).orEmpty
+
+      def renderObsBadge(obs: ObsSummary): TagMod =
+        ObsBadge(
+          obs,
+          ObsBadge.Layout.ConfAndConstraints,
```

Comment:
I think we want to show the observation's pointing name instead of the conf in this case.

Reply:
Good point. Do you think Observation name in the header and pointing in the description? Or, `conf` in the header?

We can also use actual contraints when ObsBadge is used in TargetObsList and elsewhere. But, we'll have to pull the necessary information from the API. The constraints is currently displayed as `<0.7" <0.3 mag Bright`, This would require pulling most of the constraint set from the API, or we could pull the name and use that instead. I'll look at the design doc to see if it specifies what should be displayed.

Reply:
The design shows NameAndConf. I'll go with that unless we get told otherwise.

Reply:
:+1: